src/theory/evaluator.cpp

```diff
@@ -524,6 +552,460 @@ EvalResult Evaluator::evalInternal(TNode n,
+          results[currNode] = EvalResult(val.absolute());
+          break;
+        }
+
+        case kind::FLOATINGPOINT_NEG:
+        {
+          results[currNode] = EvalResult(results[currNode[0]].d_fp.negate());
+          break;
+        }
+
+        case kind::FLOATINGPOINT_PLUS:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& lhs = results[currNode[1]].d_fp;
+          const FloatingPoint& rhs = results[currNode[2]].d_fp;
+          results[currNode] = EvalResult(lhs.plus(rm, rhs));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_SUB:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& lhs = results[currNode[1]].d_fp;
+          const FloatingPoint& rhs = results[currNode[2]].d_fp;
+          results[currNode] = EvalResult(lhs.sub(rm, rhs));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_MULT:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& lhs = results[currNode[1]].d_fp;
+          const FloatingPoint& rhs = results[currNode[2]].d_fp;
+          results[currNode] = EvalResult(lhs.mult(rm, rhs));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_DIV:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& lhs = results[currNode[1]].d_fp;
+          const FloatingPoint& rhs = results[currNode[2]].d_fp;
+          results[currNode] = EvalResult(lhs.div(rm, rhs));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_FMA:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+          const FloatingPoint& fac = results[currNode[2]].d_fp;
+          const FloatingPoint& add = results[currNode[3]].d_fp;
+          results[currNode] = EvalResult(val.fma(rm, fac, add));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_SQRT:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+          results[currNode] = EvalResult(val.sqrt(rm));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_REM:
+        {
+          const FloatingPoint& lhs = results.at(currNode[0]).d_fp;
+          const FloatingPoint& rhs = results.at(currNode[1]).d_fp;
+          FloatingPoint res = lhs.rem(rhs);
+          results[currNode] = EvalResult(lhs.rem(rhs));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_RTI:
+        {
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+          results[currNode] = EvalResult(val.rti(rm));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_MIN:
+        {
+          const FloatingPoint& lhs = results[currNode[0]].d_fp;
```

Comment:
min and max are rewritten to min and max total but ... I think that might be in expandDefintion rather than the rewriter.

Reply:
Good to know, thanks!